pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,854 @@
+package pam_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+)
+
+func ptrValue[T any](value T) *T {
+	return &value
+}
+
+func bytesPointerDecoder(ptr pam.BinaryPointer) ([]byte, error) {
+	if ptr == nil {
+		return nil, nil
+	}
+	return *(*[]byte)(ptr), nil
+}
+
+func TestSetGetItem(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(MaybeDoLeakCheck)
+
+	tests := map[string]struct {
+		item  pam.Item
+		value *string
+
+		wantValue    *string
+		wantGetError error
+		wantSetError error
+	}{
+		"Set user": {
+			item:  pam.User,
+			value: ptrValue("an user"),
+		},
+
+		"Returns empty when getting an unset user": {
+			item:      pam.User,
+			wantValue: ptrValue(""),
+		},
+
+		"Setting and getting an user": {
+			item:      pam.User,
+			value:     ptrValue("the-user"),
+			wantValue: ptrValue("the-user"),
+		},
+
+		// Error cases
+		"Error when setting invalid item": {
+			item:         pam.Item(-1),
+			value:        ptrValue("some value"),
+			wantSetError: pam.ErrBadItem,
+		},
+
+		"Error when getting invalid item": {
+			item:         pam.Item(-1),
+			wantGetError: pam.ErrBadItem,
+			wantValue:    ptrValue(""),
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(MaybeDoLeakCheck)
```

Comment:
I wonder how this work with parallel, can’t we have all tests mixing their GC failure when processing?

Reply:
Well, here we care about C-allocated data that may be still referenced or not, if that's the case when the gc is triggered then there's no error, otherwise there could be.

Indeed I had leaks around that were detected by this 100% so, it seems to behave as expected in the cases I tested.